### DISCUSSION TITLE
OAuth authentication : custom provider

### DISCUSSION THREAD

**Discussion by vd3d** [supabase/supabase#30100] [Feature Requests]
Currently, Supabase support a set of predefined oauth2 providers, but they are "almost" all the same.
behind, they just use different urls etc.

I'm looking to use OAuth with a set of other providers (not listed) and it will be very useful if we can add a set of new "custom" oauth providers from the user interface. For example, we can only provides the authorization & token urls etc... maybe some additional settings.

BTW, I have see a lot of other requests for integration with others OAuth providers (Steam, wechat, telegram, yahoo, etc etc) , this requests should solve all thoses requests.

It should be very easy to add (because you already have 99% of the codebase).

Thanks

**Comment by Revadike** [comment]
> Currently, Supabase support a set of predefined oauth2 providers, but they are "almost" all the same. behind, they just use different urls etc.
> 
> I'm looking to use OAuth with a set of other providers (not listed) and it will be very useful if we can add a set of new "custom" oauth providers from the user interface. For example, we can only provides the authorization & token urls etc... maybe some additional settings.
> 
> BTW, I have see a lot of other requests for integration with others OAuth providers (Steam, wechat, telegram, yahoo, etc etc) , this requests should solve all thoses requests.
> 
> It should be very easy to add (because you already have 99% of the codebase).
> 
> Thanks

Why not attempt to create a PR yourself? This will hopefully get the ball running. I did the exact same thing with the Steam provider support. I hope they'll take action now there is a submitted PR.

**Comment by Revadike** [comment]
[I attempted a PR for Steam provider support](https://github.com/supabase/auth/pull/1919)
However, I think the main problem is Supabase's reliability on user emails, so I am not sure if this will (fully) work.
